File: C++/Problems/Problems/problem_796.cpp
#include "problem_796.h"

bool problem_796::test() {
    std::string s = "abcde";
    std::string goal = "cdeab";

    bool expected = true;

    auto result = rotateString(s, goal);

    std::cout << result << std::endl;

    return expected == result;
}

bool problem_796::rotateString(std::string s, std::string goal) {
    int n = s.size();
    if (n != goal.size())
        return false;
    for (int i = 0; i < n; ++i)
    {
        bool match = true;
        for (int j = 0; j < n; ++j)
        {
            if (goal[j] != s[(i + j) % n])
            {
                match = false;
                break;
            }
        }
        if (match)
            return true;
    }

    return false;
}

File: Python/Problems/Problems/problem796.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem796(Problem):
    def test(self) -> bool:
        s = "abcde"
        goal = "cdeab"

        expected = True

        result = self.rotateString(s, goal)

        print(result)

        return expected == result

    def rotateString(self, s: str, goal: str) -> bool:
        n = len(s)
        if n != len(goal):
            return False
        for i in range(n):
            match = True
            for j in range(n):
                if goal[j] != s[(i + j) % n]:
                    match = False
                    break
            if match:
                return True

        return False